File: csharp/council/type_checking.cs
"polymorphism.cs"

class HighCouncilController : Controller
{
  private IJedi PrepareForMeeting(Jedi jedi)
  {
    if (jedi.IsPalpatine() || jedi.IsApprenticeOfPalpatine())
    {
      return new SithInDisguiseDecorator(jedi);
    } else if (jedi.IsSmallAndGreen()) {
      return new YodaDecorator(jedi);
    } else if (jedi is Queen) {
      return new GuestDecorator(jedi);
    } else {
      return jedi;
    }
  }
}

File: ruby/council/type_checking.rb
'polymorphism'
class HighCouncilController < ApplicationController
  def prepare_for_meeting(jedi)
    if jedi.is_palpatine? || jedi.is_apprentice_of_palpatine?
      SithInDisguisePresenter.new(jedi)
    elsif jedi.small_and_green?
      YodaPresenter.new(jedi)
    elsif jedi.class == Queen
      GuestPresenter.new(jedi)
    else
      jedi
    end
  end
end
